test/jdk/javax/swing/JProgressBar/TestProgressBarBorder.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8224261
+ * @key headful
+ * @summary Verifies if JProgressBar border is painted even though border
+ *          painting is set to false
+ * @run main TestProgressBarBorder
+ */
+
+import java.awt.Color;
+import java.awt.FlowLayout;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import javax.imageio.ImageIO;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JProgressBar;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+public class TestProgressBarBorder {
+
+    private static JFrame frame;
+    private static JProgressBar progressBar;
+    private static volatile Point pt;
+    private static volatile boolean passed;
+
+    public static void main(String[] args) throws Exception {
+        for (UIManager.LookAndFeelInfo laf :
+                UIManager.getInstalledLookAndFeels()) {
+            if (laf.getName().contains("Nimbus") || laf.getName().contains("GTK")) {
+                System.out.println("Testing LAF: " + laf.getName());
+                SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
+            } else {
+                continue;
+            }
+            Robot robot = new Robot();
+            robot.setAutoDelay(100);
+            try {
+                SwingUtilities.invokeAndWait(() -> {
+                    createAndShowUI();
+                });
+
+                robot.waitForIdle();
+                robot.delay(1000);
+
+                SwingUtilities.invokeAndWait(() -> {
+                    pt = progressBar.getLocationOnScreen();
+                });
+
+                BufferedImage borderPaintedImg =
+                        robot.createScreenCapture(new Rectangle(pt.x, pt.y,
```

Comment:
I think the test can be simplified by rendering the progress bar into the buffered image directly.

Reply:
I couldn't understand. Can you please explain a bit more?

Reply:
It is not necessary to add the progress bar to the frame and then capture the pixels by the robot. you can render content of the progress bar directly to the BufferedImage.

Reply:
There are a number of tests which do this, scaled borders are those that come to my mind first:

https://github.com/openjdk/jdk/blob/1d9688667e667dc710d64e52f1e918e047beaca3/test/jdk/javax/swing/border/EtchedBorder/ScaledEtchedBorderTest.java#L300-L311

Reply:
Another example:

https://github.com/openjdk/jdk/blob/b7425b63f6b69ba8709664377b4e037176ab6139/test/jdk/javax/swing/text/FlowView/6318524/bug6318524.java#L176-L185

Here `textPane` component is painted to the `BufferedImage`. Use the image to analyse rendering.

(In many cases, such a test can run *headless*.)

Reply:
@mrserb @aivanov-jdk Updated the test to render component to `BufferedImage` directly.